src/Microsoft.OData.Edm/Csdl/Semantics/CsdlSemanticsModel.cs

```diff
@@ -522,15 +523,40 @@ private IEdmVocabularyAnnotation WrapVocabularyAnnotation(CsdlAnnotation annotat
                 ann => new CsdlSemanticsVocabularyAnnotation(schema, targetContext, annotationsContext, ann, qualifier));
         }
 
-        private void AddSchema(CsdlSchema schema)
+        private void AddSchema(CsdlSchema schema, bool addSchemaElements = true)

```

Comment:
Would the default of adding schema elements cause any regressions in any cases and could we document how the end user would disable this incase that is the expected behaviour from their end.

Reply:
The previous behaviour was adding the schema elements by default when you called `AddSchema()`. But `AddSchema()` was only ever called if you wanted to add schema elements. With this change, `AddSchema()` can also be called when you only want to add annotations but not schema elements, or both annotations and schema elements. So setting the default for the argument to `true` retains the old behaviour while providing a way to exclude schema elements when necessary.

By the way, I'm using schema elements to refer to things like entities, enums, containers, etc. defined in the schema. I'm not sure if the variable name captures that effectively.

Reply:
Also, this method is private, and only used a few times in this class.